cockatrice/src/tab_game.cpp

```diff
@@ -376,52 +368,94 @@ TabGame::TabGame(TabSupervisor *_tabSupervisor, GameReplay *_replay)
     replayFastForwardButton->setIcon(QPixmap("theme:replay/fastforward"));
     replayFastForwardButton->setCheckable(true);
     connect(replayFastForwardButton, SIGNAL(toggled(bool)), this, SLOT(replayFastForwardButtonToggled(bool)));
-    
-    splitter = new QSplitter(Qt::Vertical);
-    splitter->addWidget(cardInfo);
-    splitter->addWidget(playerListWidget);
-    splitter->addWidget(messageLogLayoutWidget);
-
-    mainLayout = new QHBoxLayout;
-    mainLayout->addWidget(gameView, 10);
-    mainLayout->addLayout(deckViewContainerLayout, 10);
-    mainLayout->addWidget(splitter);
-    
-    QHBoxLayout *replayControlLayout = new QHBoxLayout;
+
+    replayControlLayout = new QHBoxLayout;
     replayControlLayout->addWidget(timelineWidget, 10);
     replayControlLayout->addWidget(replayStartButton);
     replayControlLayout->addWidget(replayPauseButton);
     replayControlLayout->addWidget(replayFastForwardButton);
-    
-    QVBoxLayout *superMainLayout = new QVBoxLayout;
-    superMainLayout->addLayout(mainLayout);
-    superMainLayout->addLayout(replayControlLayout);
-    
-    aNextPhase = 0;
-    aNextTurn = 0;
-    aRemoveLocalArrows = 0;
-    aRotateViewCW = 0;
-    aRotateViewCCW = 0;
-    aGameInfo = 0;
-    aConcede = 0;
-    aLeaveGame = 0;
-    aCloseReplay = new QAction(this);
-    connect(aCloseReplay, SIGNAL(triggered()), this, SLOT(actLeaveGame()));
-    
-    phasesMenu = 0;
-    gameMenu = new QMenu(this);
-    gameMenu->addAction(aCloseReplay);
-    addTabMenu(gameMenu);
-    
+
+    replayGameSceneLayout = new QVBoxLayout;
+    replayGameSceneLayout->addWidget(gameView);
+    replayGameSceneLayout->addLayout(replayControlLayout);
+
+    replayGameLayout = new QWidget;
+    replayGameLayout->setLayout(replayGameSceneLayout);
+    // end widget one
+
+    // begin widget two
+    cardInfo = new CardFrame();
+    cardHInfoLayout = new QHBoxLayout;
+    cardVInfoLayout = new QVBoxLayout;
+    cardVInfoLayout->setContentsMargins(0, 0, 0, 0);
+    cardVInfoLayout->addWidget(cardInfo);
+    cardVInfoLayout->addLayout(cardHInfoLayout);
+
+    cardReplayBoxLayoutWidget = new QWidget;
+    cardReplayBoxLayoutWidget->setLayout(cardVInfoLayout);
+    // end widget two
+
+    // begin widget three
+    playerListWidget = new PlayerListWidget(0, 0, this);
+    playerListWidget->setFocusPolicy(Qt::NoFocus);
+    messageLog = new MessageLogWidget(tabSupervisor, this);
+    // end widget two
+
+    // begin widget four
```

Comment:
It's fine if the reusability doesn't change right now - the readability will get better. And often I find that by factoring smaller I find more areas to reuse when I don't see them initially.